torch/csrc/cudnn/Conv.cpp

```diff
@@ -225,6 +225,14 @@ void cudnn_convolution_forward(
   size_t workspaceSize;
   CHECK(cudnnGetConvolutionForwardWorkspaceSize(handle, idesc.desc, wdesc.desc,
       cdesc.desc, odesc.desc, fwdAlg, &workspaceSize));
+  size_t memFree;
+  size_t memTotal;
+  cudaMemGetInfo(&memFree, &memTotal);
```

Comment:
But I thought that the caching allocator [would free all empty blocks in case of OMM, and then retry the allocation](https://github.com/pytorch/pytorch/blob/master/torch/lib/THC/THCCachingAllocator.cpp#L18-L19)?